minetest/minetest
# Improve UX when no game exists and drop `default_game`

Prerequisite of #9509 

This PR improves the UX when no game exists. If no game exists, the local tab will now point you directly at the content browser when Minetest is started (albeit with the ability to still play in multiplayer of course):

![image](https://github.com/minetest/minetest/assets/60856959/55277265-7f36-4ad8-99d8-32819f50168c)

This PR is a prerequisite of debundling MTG altogether, which has been split off and will be in another PR. Merging this PR in itself without also debundling MTG is safe, in case there is a delay in merging the MTG debundling PR.

In addition to this, the engine's behaviour when MTG doesn't exist and/or games are removed has been improved, picking the first available game (excluding devtest, for if it is on a RUN_IN_PLACE build running from source) rather than being in a weird no game void where all worlds are visible.

This also means that the `default_game` setting has been removed, as it will automatically pick the first non-devtest game. When running a dedicated server, it will print an error message pointing you to content.minetest.net to download a game to run your server with, rather than attempting to create a Minetest Game world.

This PR also fixes an issue where the games list doesn't get updated in the content tab, which is especially annoying when installing games via the "You have no games installed" message.

## To do
This PR is Ready for Review.

## How to test
1. Start out with no games installed, so you see the "no games" screen.
1. Install Minetest Game and/or some other games (at least a couple).
1. Go back and see so that a game has been automatically selected.
1. Go into the content tab and uninstall the game that got selected.
1. Go back to the local tab and see that it has selected another game, with the game theming applied and such.

Probably some more edge cases need to be tested.
